lib/pericope/parsing.rb

```diff
@@ -80,7 +80,8 @@ def match_all(text)
     end
 
     def parse_reference(book, reference)
-      parse_ranges(book, normalize_reference(reference).split(/[,;]/))
+      # Use positive lookahead to keep delimiter with the fragment that follows
+      parse_ranges(book, normalize_reference(reference).split(/(?=[,;])/))
```

Comment:
Just noting that the positive lookahead _does_ impact performance — but not by a lot

###### Before

```
PERFORMANCE
  0.021496   0.000342   0.021838 (  0.021919)
  0.191024   0.001824   0.192848 (  0.193286)
  1.908849   0.009966   1.918815 (  1.924357)
```

###### After

```
PERFORMANCE
  0.022117   0.000469   0.022586 (  0.022739)
  0.252090   0.004195   0.256285 (  0.258896)
  2.136289   0.012225   2.148514 (  2.161139)
```